python/fledge/plugins/filter/rename/rename.py

```diff
@@ -0,0 +1,180 @@
+    """ Reconfigures the plugin
+
+    Args:
+        handle:     handle returned by the plugin initialisation call
+        new_config: JSON object representing the new configuration category for the category
+    Returns:
+        new_handle: new handle to be used in the future calls
+    """
+    _LOGGER.info("Old config {} \n new config {} for {} plugin ".format(handle, new_config, PLUGIN_NAME))
+    new_handle = copy.deepcopy(new_config)
+    return new_handle
+
+
+def plugin_shutdown(handle):
+    """ Shutdowns the plugin doing required cleanup.
+
+    Args:
+        handle: handle returned by the plugin initialisation call
+    Returns:
+    """
+    global the_callback, the_ingest_ref
+    the_callback = None
+    the_ingest_ref = None
+    _LOGGER.info('{} filter plugin shutdown.'.format(PLUGIN_NAME))
+
+
+def plugin_ingest(handle, data):
+    """ Modify readings data and pass it onward
+
+    Args:
+        handle: handle returned by the plugin initialisation call
+        data:   readings data
+    """
+    global the_callback, the_ingest_ref
+    if handle['enable']['value'] == 'false':
+        # Filter not enabled, just pass data onwards
+        filter_ingest.filter_ingest_callback(the_callback, the_ingest_ref, data)
+        return
+
+    # Filter is enabled: compute for each reading
+    processed_data = {}
+    for element in data:
+        processed_data = find_and_replace(handle['operation']['value'], handle['find']['value'],
+                                          handle['replaceWith']['value'], element)
+    _LOGGER.debug("processed data {}".format([processed_data]))
+    # Pass data onwards
+    filter_ingest.filter_ingest_callback(the_callback, the_ingest_ref, [processed_data])
+
+    _LOGGER.info("{} filter ingest done".format(PLUGIN_NAME))
+
+
+def find_and_replace(operation, find, replace_with, reading):
+    """ Find and replace asset, datapoint and both with case sensitive
+    Args:
+        operation:     Possible values are asset, datapoint or both
+        find:          A regular expression to match for the given operation
+        replace_with:  A substitution string to replace the matched text with
+        reading:       A reading object
+    Returns:
+        dict:          A processed dictionary
+    """
+    new_dict = reading.copy()
+    search_pattern = r'\b{}\b'.format(find)
+    _LOGGER.debug("search_pattern: {}".format(search_pattern))
+    if operation == 'asset':
+        new_dict['asset'] = re.sub(search_pattern, replace_with, new_dict['asset'], flags=re.IGNORECASE)
+    elif operation == 'datapoint':
+        dp = re.sub(search_pattern, replace_with, str(new_dict['readings']), flags=re.IGNORECASE)
+        new_dict['readings'] = eval(dp)
+    elif operation == 'both':
+        # Both asset and datapoint case
+        both = re.sub(search_pattern, replace_with, str(reading), flags=re.IGNORECASE)
+        new_dict = eval(both)
+    else:
+        _LOGGER.warning("Unknown {} operation found.")
```

Comment:
_LOGGER.warning("Unknown operation %s, forwarding the readings as is.", operation)